File: todo/src/main/java/kr/jay/todo/domain/Todo.java
// package kr.jay.todo.domain;
//
// import java.time.LocalDateTime;
//
// import jakarta.persistence.Column;
// import jakarta.persistence.Entity;
// import jakarta.persistence.GeneratedValue;
// import jakarta.persistence.GenerationType;
// import jakarta.persistence.Id;
// import jakarta.persistence.Lob;
// import jakarta.persistence.Table;
// import lombok.*;
//
//
// @NoArgsConstructor
// @AllArgsConstructor
// @Getter
// @Builder
// @Entity
// @Table(name = "todos")
// public class Todo {
//
//     @Id
//     @GeneratedValue(strategy = GenerationType.IDENTITY)
//     private Long id;
//
//     @Column(name = "title")
//     private String title;
//
//     @Lob
//     @Column(name = "description")
//     private String description;
//
//     @Column(name = "done")
//     private Boolean done;
//
//     @Column(name = "created_at")
//     private LocalDateTime createdAt;
//
//     @Column(name = "updated_at")
//     private LocalDateTime updatedAt;
//
//     public void update(String title, String description, Boolean done) {
//         this.title = title;
//         this.description = description;
//         this.done = done != null && done;
//         this.updatedAt = LocalDateTime.now();
//     }
// }

File: todo/src/main/java/kr/jay/todo/domain/Todo.kt
package kr.jay.todo.domain

import jakarta.persistence.*
import java.time.LocalDateTime

/**
 * Todo
 *
 * @author jaypark
 * @version 1.0.0
 * @since 2023/08/17
 */
@Entity
@Table(name = "todos")
class Todo(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id : Long? = 0,

    @Column(name ="title")
    var title : String,

    @Lob
    @Column(name = "description")
    var description : String,

    @Column(name = "done")
    var done : Boolean,

    @Column(name = "created_at")
    var createdAt : LocalDateTime,

    @Column(name = "updated_at")
    var updatedAt : LocalDateTime? = null
) {

    fun update(
        title: String,
        description: String,
        done: Boolean
    ) {
        this.title = title
        this.description = description
        this.done = done
        this.updatedAt = LocalDateTime.now()
    }
}